snapcraft/parts/plugins/matter_plugin.py

```diff
@@ -0,0 +1,173 @@
+            "ninja-build",
+            "libssl-dev",
+            "libdbus-1-dev",
+            "libglib2.0-dev",
+            "libavahi-client-dev",
+            "python3-venv",
+            "python3-dev",
+            "python3-pip",
+            "libgirepository1.0-dev",
+            "libcairo2-dev",
+            "libreadline-dev",
+            "generate-ninja",
+        }
+
+    @overrides
+    def get_build_environment(self) -> Dict[str, str]:
+        return {}
+
+    @overrides
+    def get_build_snaps(self) -> Set[str]:
+        return set()
+
+    @overrides
+    def get_build_commands(self) -> List[str]:
+        options = cast(MatterPluginProperties, self._options)
+        commands = []
+
+        if self.snap_arch == "arm64":
+            commands.extend(
+                [
+                    f"wget --no-verbose https://github.com/project-chip/zap/releases/download/"
+                    f"{options.matter_zap_version}/zap-linux-{self.snap_arch}.zip",
+                    f"unzip -o zap-linux-{self.snap_arch}.zip",
+                    "echo 'export ZAP_INSTALL_PATH=$PWD'",
+                ]
+            )
+
+        # Clone Matter repository if not present
+        commands.extend(
+            [
+                f"if [ ! -d matter ]; then git clone --depth 1 -b {options.matter_sdk_version} "
+                f"{MATTER_REPO} matter && cd matter; else cd matter || echo 'skip clone'; fi"
+            ]
+        )
+
+        # Checkout submodules for Linux platform
+        commands.extend(["scripts/checkout_submodules.py --shallow --platform linux"])
+
+        """ 
+        The project writes its data to /tmp which isn't persisted.
+
+        Setting TMPDIR env var when running the app isn't sufficient as 
+        chip_[config,counter,factory,kvs].ini still get written under /tmp.
+        The chip-tool currently has no way of overriding the default paths to
+        storage and security config files.
+
+        Snap does not allow bind mounting a persistent directory on /tmp, 
+        so we need to replace it in the source with another path, e.g. /mnt.
+        See the top-level layout definition which bind mounts a persisted
+        directory within the confined snap space on /mnt.
+        """
+        """Replace storage paths"""
+        commands.extend(
+            [
+                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPLinuxStorage.h",
+                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h",
+            ]
+        )
+
+        """
+        Bootstrapping script for building Matter SDK with minimal "build" requirements 
+        and setting up the environment
+        """
+        commands.extend(
+            ["set +u && source setup/bootstrap.sh --platform build && set -u"]
+        )
+
+        commands.extend(
+            [
+                "cp -vr ./* $CRAFT_PART_INSTALL/",
+                "echo 'Cloned Matter repository and built Matter SDK'",
+            ]
+        )
```

Comment:
Do you really need copy everything to the part? I am not sure `bootstrap.sh` doing but there is not a finally result to use?

Reply:
Good catch. I think nothing should be staged here. Only the application's binary distribution, built by the consumer snap needs to be staged.

Reply:
Thanks. You are right, the copy is redundant. It has been removed in the latest commit.